#208 Updating readme to add badge

Conor0Callaghan opened the pull request:
Add badge to README as per #208 part 1

Conor0Callaghan committed:
#208 Updating readme to add badge

Add badge to README as per #208 part 1

rohit-ganguly reviewed (commented):
Thanks for adding the shield, that's awesome. I think it would be useful to add a short section around installation via pip/pipenv too towards the start of the doc!

tianyizheng02 reviewed (changes requested):
For convenience, is it possible to have the badge be a clickable link to PittAPI's PyPI page?

Conor0Callaghan committed:
Merge branch 'pittcsc:dev' into dev

Conor0Callaghan committed:
Update README.md

Updating Readme with first attempt at additional install commands as per comments in #208

Conor0Callaghan commented:
> Thanks for adding the shield, that's awesome. I think it would be useful to add a short section around installation via pip/pipenv too towards the start of the doc!

hopefully that's a good start!

tianyizheng02 commented on README.md:
```diff
@@ -3,9 +3,18 @@
 
+## Installation
+
+Pitt API can be installed using the python package manager ``pip``. To make your development easier, you can run the pip commands in a virtual environment. 
```
```suggestion
The Pitt API can be installed using the Python package manager `pip`. To make your development easier, you can run the pip commands in a virtual environment. 
```

tianyizheng02 reviewed (dismissed):
Personally I don't think we need to include instructions for setting up a virtual environment in the Installation section. I think it's better to leave that choice of using a virtual environment (and how to set one up) up to the user, rather than recommending it with `pipenv` specifically. I feel that it's enough to explain how to install the package itself through `pip`.

timparenti commented:
> Personally I don't think we need to include instructions for setting up a virtual environment in the Installation section. I think it's better to leave that choice of using a virtual environment (and how to set one up) up to the user

Since many of the intended audience for this API are beginners and learners, I disagree.  I think it's important to explicitly let folks at that level know of the option so that they can make an informed decision on whether they want to "pollute" their default dependency tree, or at least be aware that that could happen in the likely case they want to keep something else separate later.

Now, could this text maybe be a bit clearer that a virtual environment is not a _requirement_, but merely a _best practice_ that's worth considering?  Absolutely.  Should it instead just point people to good write-ups on how to set up the most common types instead of providing step-by-step instructions for one option ourselves?  Perhaps.  But, to my mind, that adds a dose of unnecessary friction, and for something this simple, I don't think the proposed instructions are harmful as written.

Beginners have to be introduced to these concepts somehow so they can experiment and, as is your goal, ultimately make their own choices.  I think this project is a _fantastic_ low-stakes place for folks to encounter and weigh these considerations, if they haven't before.

azharichenko commented on README.md:
```diff
@@ -3,9 +3,18 @@
+Pitt API can be installed using the python package manager ``pip``. To make your development easier, you can run the pip commands in a virtual environment. 
+
+1. Install ``pipenv`` using the [instructions](https://pipenv.pypa.io/en/latest/installation.html) for your operating system, pipenv is a tool which manages your python virtual environments for you. 
+2. Run the commands ``pipenv install`` and ``pipenv shell`` to create and setup the virtual environment.
```
```suggestion
2. Run the commands ``pipenv sync --dev`` and ``pipenv shell`` to create and setup the virtual environment.
```

azharichenko committed:
Update README.md

azharichenko committed:
Update README.md

Co-authored-by: Tianyi Zheng <tianyizheng02@gmail.com>

azharichenko committed:
Merge branch 'dev' into dev
